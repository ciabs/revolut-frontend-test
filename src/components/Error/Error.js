import React from 'react';
import FontAwesome  from 'react-fontawesome';
import PropTypes from 'prop-types';

export const Error = error => (
  <FontAwesome
    style={{
      color: 'red',
      position: 'absolute',
      top: '20px',
      right: '20px',
      display: error.error ? 'block': 'none',
      cursor: 'pointer'
    }}
    name='exclamation-triangle'
    size='2x'
    title={error.error}
  />
);

Error.propTypes = {
  error: PropTypes.string
};
