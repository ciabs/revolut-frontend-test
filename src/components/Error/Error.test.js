import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {Error} from './Error';

describe('Error', () => {
  const props = {
    error: null
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Error {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Error {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
