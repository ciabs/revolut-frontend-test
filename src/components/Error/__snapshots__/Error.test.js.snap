// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error has a valid snapshot 1`] = `
<span
  aria-hidden={true}
  className="fa fa-exclamation-triangle fa-2x"
  style={
    Object {
      "color": "red",
      "cursor": "pointer",
      "display": "none",
      "position": "absolute",
      "right": "20px",
      "top": "20px",
    }
  }
  title={null}
/>
`;
